raiden-ts/src/raiden.ts

```diff
@@ -1011,6 +1021,53 @@ export class Raiden {
         .reduce((acc, iou) => acc.add(iou.amount), Zero);
     return balance.sub(owedAmount);
   }
+
+  /**
+   * Deposits the amount to the UserDeposit contract with the target/signer as a beneficiary.
+   * The deposited amount can be used as a collateral in order to sign valid IOUs that will
+   * be accepted by the Services.
+   *
+   * Throws an error, in the following cases:
+   * <ol>
+   *  <li>The amount specified equals to zero</li>
+   *  <li>The target has an insufficient token balance</li>
+   *  <li>The "approve" transaction fails with an error</li>
+   *  <li>The "deposit" transaction fails with an error</li>
+   * </ol>
+   *
+   * @param amount - The amount to deposit on behalf of the target/beneficiary.
+   * @returns transaction hash
+   */
+  public async depositToUDC(amount: BigNumberish): Promise<Hash> {
+    const depositAmount = bigNumberify(amount);
+
+    if (depositAmount.isZero()) throw new Error('Please deposit a positive amount.');
+
+    const { userDepositContract, address } = this.deps;
+
+    const tokenAddress = await this.userDepositTokenAddress();
+    const serviceToken = CustomTokenFactory.connect(tokenAddress, this.deps.signer);
+    const balance = await serviceToken.functions.balanceOf(address);
+
+    if (balance.lt(amount)) throw new Error(`Insufficient token balance (${balance}).`);
+
+    const approveTx = await serviceToken.functions.approve(
+      userDepositContract.address,
+      depositAmount,
+    );
+    const approveReceipt = await approveTx.wait();
```

Comment:
If you're confident checks above (i.e. balance) are enough to ensure the tx will go through, a nice optimization is to move `wait`ing for tx to be mined after deposit tx is subtmitted (you can then wait on `Promise.all()` txs). This way, both txs are queued, and can even be mined on the same block, as ethereum node is smart enough to queue correctly by nonce. This is an optimization I plan to implement for `channelDeposit` as well, but this could be a nice demo of this approach. Worse case you just spend a little more of gas, but the speed gains in the trivial case may be worth it if the unhappy case is unlikely/impossible.

Reply:
You can never be sure. But I can give it a try and see how it works.

Reply:
```typescript
  const [approveReceipt, depositReceipt] = await Promise.all([approveTx.wait(), depositTx.wait()])
``` 
This is what you had in mind right?